codec-http/src/main/java/io/netty/handler/codec/http/websocketx/extensions/compression/DeflateDecoder.java

```diff
@@ -81,12 +114,14 @@ protected void decode(ChannelHandlerContext ctx, WebSocketFrame msg, List<Object
         }
 
         boolean readable = msg.content().isReadable();
+        boolean emptyDeflateBlock = EMPTY_DEFLATE_BLOCK.equals(msg.content());
```

Comment:
maybe we can inline this into the condition where it is used?

`if (readable && compositeDecompressedContent.numComponents() <= 0 && !EMPTY_DEFLATE_BLOCK.equals(msg.content())) {`

Reply:
```
decoder.writeInbound(msg.content().retain());
        if (appendFrameTail(msg)) {
            decoder.writeInbound(FRAME_TAIL.duplicate());
        }
```
after decompression msg.content() is not readable.